### PR TITLE
[FIX] web, *: tours correctly break when there are missing deps

### PR DESCRIPTION
**Pull request by sdegueldre**
*: im_livechat, website_livechat, website_slides

In https://github.com/odoo/odoo/commit/c989ff339d1bb8afde3371d12dab8cc227941691, the odoo module system has been changed so that it now
errors out when modules are defined but have missing dependencies and
cannot run. Unfortunately, tours are defined in a single bundle,
`assets_tests`, and this bundle is used everywhere tours need to run,
but some of these tours depend on modules that only exist in the backend
and others on modules that only exist in the frontend. We don't want to
break up the `assets_tests` bundle as this creates a poor developer
experience, and so the solution that was chosen at the time is that when
the `assets_tests` bundle is defined, we ignore all module dependency
errors.

This causes a problem however, which is that if a module dependency is
missing in production code, the page will be broken by the module error
(which is expected) but if the `assets_tests` are present, which is the
case when running tours and tests, then it won't, meaning the tours will
happily pass despite the application being broken, which is
unacceptable.

This commit fixes this issue by replacing the assets_test bundle by a
new bundle (`__assets_tests_call__`) within which all module dependency
errors are ignored, which lets dependency errors that are hapenning in
other bundles (ie in production code) happen even during the tests.

Because of this, this commit also fixes the dependency errors that
already existed but were previously silent:

- for im_livechat and website_livechat, the code was refactored to use
the standard mechanism that we use elsewhere to add data directly in the
page: we store the data in __session_info__, and the module that exposes
that data is always present, instead of having a module that's defined
conditionally, which causes issues when non-conditional modules depend
on it.
- in web, the `report_assets_common` bundle was used alongside
`assets_common` but without `assets_frontend` or `assets_backend`.
Reports do not need any js and it appears that it already had what it
needed as far as css is concerned except for one fontawesome override.
Code was changed so that the js from the bundle is not included and it
reports no longer use `assets_common`, the js part of the bundle is no
longer used.
- in website_slides, the `slide_embed_assets` bundle is used alongside
`assets_common` without using `assets_frontend` or `assets_backend`
which is required for `assets_common` to function, the approach here was
to simply take the files from `assets_common` and remove the files that
were causing issues. This bundle should be trimmed at some point, as it
contains way too many things that it does not need.

The file colors.js was also moved to core, as it isn't graph specific
and was causing dependency issues with a field that was importing it.

### CODE REVIEW

**Comment by sdegueldre** [comment]
@odoo/rd-website Hello, can you take a look at the changes in website_slides? I've taken the same "shotgun approach" as what was taken when you stopped using assets_common in the frontend of just copy-pasting the contents of assets common into the slide_embed_assets bundle, to keep it similar in terms of behaviour as what is going on right now, but I'm honestly confused why it was using assets_common's js at all, considering the pdf embed widget it uses doesn't depend on odoo modules at all. So the bundle can probably be trimmed substantially but I'm not exactly sure if there's a reason it was using assets_common that I'm just not aware of.

**Comment by qsm-odoo** [comment]
@sdegueldre 

> I'm not exactly sure if there's a reason it was using assets_common that I'm just not aware of.

Me neither, you should probably ask the team that handles that app (TDE / SM team), thanks :+1:

**Comment by sdegueldre** [comment]
@odoo/rd-sm-learn Whoops, pinged the wrong team regarding website_slides. Any idea why the website_slides embed assets bundle uses assets common?

**Comment by tde-banana-odoo** [comment]
> > @odoo/rd-sm-learn Whoops, pinged the wrong team regarding website_slides. Any idea why the website_slides embed assets bundle uses assets common?
> 
> Hello,
> 
> It seems you did it at [88c2b68](https://github.com/odoo/odoo/commit/88c2b6843a09ba087a10388fc39cb69dac4e2949) (ok, quite old). Main reason why it is still looking like that is that we don't really use that feature and/or plan to really promote its usage. We just somehow support that embed mode with minimal bugfixes when it seems broken.
> 
> I think a few people want to keep it and that's is mainly why it is still there. This is not really an interesting feature of eLearning module. I think it should be completely removed, as it makes no sense to allow including Odoo pages in external websites. No other Odoo module does that, except maybe livechat, and I am not sure it still works. But that's not part of our scope to do that kind of feature. But anyway keeping or removing this feature is outside the scope of this discussion :) .
> 
> TL;DR: don't know, feel free to clean it, our position being we support and hope to remove it.
> 
> Cheers,

**Comment by tde-banana-odoo** [comment]
(sorry, wrong PR management :D )

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Comment by qsm-odoo** [comment]
> It seems you did it at https://github.com/odoo/odoo/commit/88c2b6843a09ba087a10388fc39cb69dac4e2949 (ok, quite old)

Indeed, sorry, my memory does not go that far I guess :grimacing: But even if I remembered, I would have let you take the decision about it now that team reviews/codeowners make more sense than in 2016 when I probably merged that by myself without any review ^^

Note that from what I gather from my commit what I did is replace the inclusion of bootstrap JS, font awesome CSS and jQuery JS by a call to assets_common which would now contain that and some small utils from web. May indeed not have been the best choice since assets_common kept growing since then and we are coming back to one assets everywhere now.

Anyways, I see this was r+'ed but did not you make a mistake? If I understand this well you added all the files CSS + JS of assets_common directly inside the slide_embed_assets but you are still using t-call-assets of assets_common for the CSS part, won't you have the CSS two times @sdegueldre ? :thinking:

**Comment by tde-banana-odoo** [comment]
> > It seems you did it at [88c2b68](https://github.com/odoo/odoo/commit/88c2b6843a09ba087a10388fc39cb69dac4e2949) (ok, quite old)
> 
> Indeed, sorry, my memory does not go that far I guess grimacing But even if I remembered, I would have let you take the decision about it now that team reviews/codeowners make more sense than in 2016 when I probably merged that by myself without any review ^^

Those were hard times :D well my point was more that I don't know much more than you about the embed mode implementation, so if you feel like it is cleaner with this patch than previously, don't hesitate :) . I will create a task to cleanup that stuff sooner or later, and check if we cannot remove it.

> 
> Note that from what I gather from my commit what I did is replace the inclusion of bootstrap JS, font awesome CSS and jQuery JS by a call to assets_common which would now contain that and some small utils from web. May indeed not have been the best choice since assets_common kept growing since then and we are coming back to one assets everywhere now.

Indeed :) . Good idea to keep an eye on that.

> 
> Anyways, I see this was r+'ed but did not you make a mistake? If I understand this well you added all the files CSS + JS of assets_common directly inside the slide_embed_assets but you are still using t-call-assets of assets_common for the CSS part, won't you have the CSS two times @sdegueldre ? thinking

:eyes:
